tests/conftest.py

```diff
@@ -115,18 +120,37 @@ def clean_serveradmin(filters):
     Query(filters).delete().commit()
 
 
+def clean_aws(vm_hostname):
+    try:
+        vm_stop(vm_hostname)
+        vm_delete(vm_hostname)
+    except (VMError, DatasetError):
+        # No object found or no cleanup needed
+        pass
+
+
 def get_next_address(vm_net, index):
-    subnet_levels = ceil(log(PYTEST_XDIST_WORKER_COUNT, 2))
+    non_vm_hosts = list(Query({
+        'project_network': vm_net,
+        'servertype': Not('vm'),
+    }, ['intern_ip']))
+    offset = 1 if len(non_vm_hosts) > 0 else 0
+    subnet_levels = ceil(log(PYTEST_XDIST_WORKER_COUNT + offset, 2))
     project_network = Query({'hostname': vm_net}, ['intern_ip']).get()
-
     try:
-        subnets = project_network['intern_ip'].subnets(subnet_levels)
+        subnets = list(project_network['intern_ip'].subnets(subnet_levels))
     except ValueError:
         raise Exception(
             'Can\'t split {} into enough subnets '
             'for {} parallel tests'.format(
                 vm_net, PYTEST_XDIST_WORKER_COUNT,
             )
         )
-
-    return list(subnets)[PYTEST_XDIST_WORKER][index]
+    if len(non_vm_hosts) > subnets[0].num_addresses:
+        raise Exception(
+            'Can\'t split {} into enough subnets '
+            'for {} parallel tests'.format(
+                vm_net, PYTEST_XDIST_WORKER_COUNT,
+            )
+        )
+    return list(subnets)[PYTEST_XDIST_WORKER + 1][index]
```

Comment:
i know i suggested this to you in the first place but we don't actually need the `list()` here again. it's already done above when assigning it to the `subnets` variable